### DISCUSSION TITLE
Differences in downloads of Firmware for Adventurer 5M - Version 2.4.5

### DISCUSSION THREAD

**Discussion by DougJoseph** [xblax/flashforge_ad5m_klipper_mod#152] [General]
When I download and uncompress the Zip file from the Flashforge support download site ( https://www.flashforge.com/download-center/143 ), it results in a folder named "Adventurer5M-2.4.5-2.2.3-20240118" — and that folder has 13 files:

- boot.img
- control-2.2.3.tar.xz
- Cube-PLA-Test.gx
- end.img
- flashforge_init.sh
- inittab
- kernel-2.1.4.tar.xz
- library-2.1.0.tar.xz
- mencoder
- play
- shadow
- software-2.4.5.tar.xz
- start.img

However, by contrast, the previous firmware file I obtained from them somehow, which came as a TGZ file, when I uncompress it, it results in a folder named "Adventurer5M-2.4.5-2.2.3-20240118-Factory" — and that folder has 18 files:

- boot.img
- control-2.2.3.tar.xz
- Cube-PLA-Test.gx
- end.img
- flashforge_init.sh
- inittab
- kernel-2.1.4.tar.xz
- library-2.1.0.tar.xz
- mencoder
- play
- printer.base.cfg
- printer.cfg
- rtl_hostapd_2G.conf
- screwflag
- shadow
- software-2.4.5.tar.xz
- start.img
- tmc.py

The 5 files that are present in the TGZ file, but that are missing from the Zip file are:

- printer.base.cfg
- printer.cfg
- rtl_hostapd_2G.conf
- screwflag
- tmc.py

Also, note the name differences:
From Zip: "Adventurer5M-2.4.5-2.2.3-20240118"
From TGZ: "Adventurer5M-2.4.5-2.2.3-20240118-Factory"

I am attaching both files here, in the event it helps anyone, especially given that I currently cannot find where to again download the TGZ file I was provided at some point. I am wrapping the TGZ file as a Zip file, to get it small enough to attach here. Simply uncompress that Zip to get the TGZ file out of it.

[Adventurer5M-2.4.5-2.2.3-20240118-Factory.tgz.zip](https://github.com/xblax/flashforge_ad5m_klipper_mod/files/15047125/Adventurer5M-2.4.5-2.2.3-20240118-Factory.tgz.zip)

[Adventurer5M-2.4.5-2.2.3-20240118.zip](https://github.com/xblax/flashforge_ad5m_klipper_mod/files/15047130/Adventurer5M-2.4.5-2.2.3-20240118.zip)

**Comment by consp** [comment]
> printer.base.cfg
> printer.cfg
> rtl_hostapd_2G.conf
> screwflag
> tmc.py

They are all already present on the system anyway. My guess would be they use different script (or a "manual script" aka human) to create one of the files and just grabbed everything in the directory with the tgz file.

With the "new" 2.6.x release the `rtl_hostapd_2G.conf` file was updated so they are not that static.

The digest for all files are the same (if included) except for the bootscript (`flashforge_init.sh`) which is to be expected since it contains a check for the name of the printer.

**Comment by ugryy** [comment]
Oh no, I made a mistake for sake of installing Klipper mode on my 5M Pro.
I tried to flash klipper according to install guide but printer said something like "this version is not compatible with firmware". I had 2.6.x firmware installed. 
Then I tried to downgrade to 2.4.5, and found this discussion. But didn't pay attention to the name it was non pro at the moment. Now my printer is 5m firmware installed pro lol. I can't install 5m pro firmwares on flashforge's website. Tried to change names etc. but nope. 
I emailed flashforge but don't know what to do now. I could use some help! Thanks.

**Reply by consp** [reply]
I'm not 100% sure I understand but if you (somehow) installed the flashforge firmeware for the "normal" on a pro, you can do the following:

* Download the pro firmware .tgz from the flashforge site
* Clean the usb drive (remove all files)
* Extract everything in the .tgz file to a usb drive, do *not* copy the file but extract it
* Edit `flashforge_init.sh` on the usb drive and alter `MACHINE=Adventurer5MPro` to `MACHINE=Adventurer5M` and `PID=0024` to `PID=0023`
* Insert the USB drive and let it install
* If it fails, try all `MACHINE=` and `PID=` combinations

**Reply by ugryy** [reply]
Thank you so much for helping, support send me a 2.4.5 5m pro factory firmware and it worked I'm so happy. The information u gave me was what I needed if support didn't sent me the fw. Now I will try to install mainsail! 
If someone need the factory 2.4.5 fw, let me know.

**Reply by flyespresso** [reply]
@ugryy  I'd love a copy of it just as a fallback in case I need it with my pro!

**Reply by ugryy** [reply]
Sure! Here you go:
[Uploading Adventurer5MPro-2.4.5-2.2.3-20240118-Factory (1).tgz.zip…]()

**Reply by system11b** [reply]
The link above just comes back to this discussion, could you post the zip please?

**Reply by consp** [reply]
> The link above just comes back to this discussion, could you post the zip please?

This happens when the upload was not finished, timed out or was too large. Better put it in a file sharing medium if it's big.

**Reply by eduncan911** [reply]
The download links are archived here: https://github.com/xblax/flashforge_ad5m_klipper_mod/issues/211#issuecomment-2195816779
